### ISSUE TITLE
taoseq v2.0.1 is broken with permutate

### ISSUE THREAD

**Issue by bcharlier** [getkeops/keops#115]
Hi @fradav ,

switching to branch taoseq2 (I just update the sequence submodule to v2.0.1) and try to compile `test_tensordot_formula_cpp` we get the following error (g++-9 or g++-10): 
```
Scanning dependencies of target test_tensordot_formula_cpp
[ 50%] Building CXX object examples/CMakeFiles/test_tensordot_formula_cpp.dir/test_tensordot_formula.cpp.o
In file included from /home/bcharlier/projets/keops/keops/keops/keops_includes.h:85,
                 from /home/bcharlier/projets/keops/keops/keops/examples/test_tensordot_formula.cpp:7:
/home/bcharlier/projets/keops/keops/keops/core/formulas/maths/TensorDot.h: Dans l'instanciation de « struct keops::TensorDot<keops::Var<0, 432, 0>, keops::Var<1, 144, 1>, std::integer_sequence<long unsigned int, 3, 12, 12>, std::integer_sequence<long unsigned int, 12, 12>, std::integer_sequence<long unsigned int, 1, 2>, std::integer_sequence<long unsigned int, 0, 1> > » :
/home/bcharlier/projets/keops/keops/keops/core/pre_headers.h:43:8:   requis par « struct keops::KeopsNS<keops::TensorDot<keops::Var<0, 432, 0>, keops::Var<1, 144, 1>, std::integer_sequence<long unsigned int, 3, 12, 12>, std::integer_sequence<long unsigned int, 12, 12>, std::integer_sequence<long unsigned int, 1, 2>, std::integer_sequence<long unsigned int, 0, 1> > > »
/home/bcharlier/projets/keops/keops/keops/examples/test_tensordot_formula.cpp:47:12:   requis depuis ici
/home/bcharlier/projets/keops/keops/keops/core/formulas/maths/TensorDot.h:267:4: erreur: l'expression « std::integer_sequence<long unsigned int, 0>::size » n'est pas une expression constante
   >;
    ^
```
See comment line: https://github.com/getkeops/keops/blob/520418eb3bdc22628f8c7aee9a809bbb54fb87db/keops/core/formulas/maths/TensorDot.h#L256

Note: the cuda version (namly `test_tensordot_formula`) do not work either. But the error is different. My understanding is that we should fix the cpp version first.

**Comment by fradav** [comment]
A wild guess is that something related to the direct use of std::integer_sequence in tao:seq (which provides its own integer_sequence wrapper) changed with the update, investigate it.

**Comment by fradav** [comment]
Fixed in ee0c41d
But NOT for cuda 11.1

**Comment by bcharlier** [comment]
fix in 1af2124
